src/assertions.jl

```diff
@@ -0,0 +1,41 @@
+"""
+    Assertion
+
+An Assertion stores settings for checking the validity of a `AbstractArray` or `Tables.table` containing missing values.
+New assertions are expected to subtype `Impute.Assertion` and, at minimum,
+implement the `assert(data::AbstractVector{Union{T, Missing}}, ::<MyAssertion>)` method.
+"""
+abstract type Assertion end
+
+
+"""
+    assert(data, a::Assertion; kwargs...)
+
+If the assertion `a` fails then an `AssertionError` is thrown, otherwise the `data`
+provided is returned without mutation.
+
```

Comment:
```suggestion

Implictly broadcasts over columns of a table,
and over <...explain what it does to matrix ...:
```

Reply:
Hmm, the table method has its own docstring that explains how the assertion is being applied columnwise on tables. I feel like the behaviour n-dim arrays is explained by thte keyword?